indexing-service/src/main/java/org/apache/druid/indexing/overlord/ImmutableWorkerInfo.java

```diff
@@ -225,6 +270,79 @@ private int getWorkerParallelIndexCapacity(double parallelIndexTaskSlotRatio)
     return workerParallelIndexCapacity;
   }
 
+  private boolean hasSufficientWorkerCapacity(Task task)
+  {
+    int capacityRemaining = worker.getCapacity() - getCurrCapacityUsed();
+    int requiredCapacity = task.getTaskResource().getRequiredCapacity();
+    return capacityRemaining >= requiredCapacity;
+  }
+
+  private boolean isAvailabilityGroupAvailable(Task task)
+  {
+    return !getAvailabilityGroups().contains(task.getTaskResource().getAvailabilityGroup());
+  }
+
+  private boolean canRunTaskBasedOnCustomLimit(Task task, Map<String, Integer> limitsMap, Map<String, Double> ratiosMap)
```

Comment:
This method will become simpler once we remove the absolute limits.

Reply:
Simplified